internal/graph/lookupsubjects.go

```diff
@@ -460,7 +464,10 @@ func (lsu *lookupSubjectsUnion) CompletedChildOperations() error {
 
 		for _, result := range collector.Results() {
 			metadata = combineResponseMetadata(metadata, result.Metadata)
-			foundSubjects.UnionWith(result.FoundSubjectsByResourceId)
+			err := foundSubjects.UnionWith(result.FoundSubjectsByResourceId)
+			if err != nil {
+				return err
```

Comment:
Wrap these errors with higher level context about _why_ you were trying to Union, so we can trace which callsite caused the problem.